src/sentry/remote_config/docs/api.md

```diff
@@ -53,105 +39,231 @@ Retrieve the project's configuration.
       "options": {
         "sample_rate": 1.0,
         "traces_sample_rate": 0.5
+      },
+      "version": 1
+    }
+    ```
+
+## Configurations [/projects/<organization_id_or_slug>/<project_id_or_slug>/configurations/]
+
+| Field              | Type             | Description                                                                   |
+| ------------------ | ---------------- | ----------------------------------------------------------------------------- |
+| environment        | optional[string] | The environment the configuration is associated with.                         |
```

Comment:
Does this field need to be unique across all the configs inside a given project? 

That opens a can of worms if not... like flattening configs or something idk... there's no order so flattening here isn't gonna work

Reply:
There should be a unique constraint on `(project_id, environment, name)`.  Definitely no merging/flattening.

Reply:
`environment` `id` and `name` also seem to be a trio to look at. 

Maybe we should be looking at a different url scheme:
1. `GET /projects/<organization_id_or_slug>/<project_id_or_slug>/configurations/` returns a list of envs basically (still the same config objects, but the unique constraint is built-in)
2. Rename other endpoint from `<configuration_id>` to `<env>`: `GET /projects/<organization_id_or_slug>/<project_id_or_slug>/configurations/<env>`
2. Drop `name` field from inside the body. It's basically the same as concatenating project+env. Could be added back later.
3. Drop `id` field from inside the body. Unless it's really needed by the db (and it could be!) but it's already replaced by the url at this point.

still have to deal with the `null` environment example. But that could be where merging comes back into play. What's better is that it serves as a special default/fallback.

I think on the topic of env names changing... we shouldn't try to be smart about that. People should be able to create a config for an env that doesn't exist yet. And if the SDK stops sending data to an env we shouldn't move the config or do anything with it either. Instead we can show volume of requests against an env to let people delete things that are unused if needed.